File: app/src/main/java/com/cvgenerator/ui/screens/PreviewScreen.kt
package com.cvgenerator.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.cvgenerator.data.CvViewModel
import com.cvgenerator.data.Skill
import com.cvgenerator.ui.theme.Cream
import com.cvgenerator.ui.theme.DeepBlack
import com.cvgenerator.ui.theme.Gold
import com.cvgenerator.ui.theme.MediumGray

@Composable
fun PreviewScreen(viewModel: CvViewModel, onExportPdf: () -> Unit) {
    val cvData by viewModel.cvData.collectAsState()
    val info = cvData.personalInfo

    Column(
        modifier = Modifier
            .fillMaxSize()
            .verticalScroll(rememberScrollState())
    ) {
        // CV Preview — white card simulating A4
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(12.dp)
                .background(Cream, RoundedCornerShape(8.dp))
                .padding(24.dp)
        ) {
            // Header block
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .background(DeepBlack, RoundedCornerShape(6.dp))
                    .padding(20.dp)
            ) {
                Text(
                    info.fullName.ifBlank { "Votre Nom" },
                    fontSize = 22.sp,
                    fontWeight = FontWeight.Bold,
                    color = Cream
                )
                Text(
                    info.jobTitle.ifBlank { "Titre professionnel" },
                    fontSize = 14.sp,
                    color = Gold,
                    fontWeight = FontWeight.Medium
                )
                Spacer(Modifier.height(8.dp))
                Row(horizontalArrangement = Arrangement.spacedBy(16.dp)) {
                    if (info.email.isNotBlank()) CvChip(info.email)
                    if (info.phone.isNotBlank()) CvChip(info.phone)
                    if (info.city.isNotBlank()) CvChip(info.city)
                }
            }

            Spacer(Modifier.height(16.dp))

            // Summary
            if (info.summary.isNotBlank()) {
                CvSection("Profil") {
                    Text(info.summary, fontSize = 13.sp, color = Color(0xFF333333), lineHeight = 20.sp)
                }
            }

            // Experience
            if (cvData.experiences.isNotEmpty()) {
                CvSection("Expériences") {
                    cvData.experiences.forEach { exp ->
                        Column(modifier = Modifier.padding(bottom = 12.dp)) {
                            Row(
                                modifier = Modifier.fillMaxWidth(),
                                horizontalArrangement = Arrangement.SpaceBetween
                            ) {
                                Text(exp.position, fontWeight = FontWeight.SemiBold, fontSize = 13.sp, color = DeepBlack)
                                Text(
                                    "${exp.startDate} — ${if (exp.isCurrent) "Présent" else exp.endDate}",
                                    fontSize = 11.sp, color = Color(0xFF666666)
                                )
                            }
                            Text(exp.company, fontSize = 12.sp, color = Color(0xFF444444), fontWeight = FontWeight.Medium)
                            if (exp.description.isNotBlank()) {
                                Text(exp.description, fontSize = 12.sp, color = Color(0xFF555555), lineHeight = 18.sp)
                            }
                        }
                    }
                }
            }

            // Education
            if (cvData.educations.isNotEmpty()) {
                CvSection("Formation") {
                    cvData.educations.forEach { edu ->
                        Column(modifier = Modifier.padding(bottom = 10.dp)) {
                            Row(
                                modifier = Modifier.fillMaxWidth(),
                                horizontalArrangement = Arrangement.SpaceBetween
                            ) {
                                Text(edu.degree, fontWeight = FontWeight.SemiBold, fontSize = 13.sp, color = DeepBlack)
                                Text("${edu.startYear} — ${edu.endYear}", fontSize = 11.sp, color = Color(0xFF666666))
                            }
                            Text(edu.school, fontSize = 12.sp, color = Color(0xFF444444))
                            if (edu.field.isNotBlank()) Text(edu.field, fontSize = 12.sp, color = Color(0xFF666666))
                        }
                    }
                }
            }

            // Skills
            if (cvData.skills.isNotEmpty()) {
                CvSection("Compétences") {
                    cvData.skills.chunked(2).forEach { row ->
                        Row(modifier = Modifier.fillMaxWidth().padding(bottom = 8.dp), horizontalArrangement = Arrangement.spacedBy(12.dp)) {
                            row.forEach { skill ->
                                SkillBar(skill = skill, modifier = Modifier.weight(1f))
                            }
                            if (row.size == 1) Spacer(Modifier.weight(1f))
                        }
                    }
                }
            }

            // Languages
            if (cvData.languages.isNotEmpty()) {
                CvSection("Langues") {
                    Row(horizontalArrangement = Arrangement.spacedBy(16.dp)) {
                        cvData.languages.forEach { lang ->
                            Text("${lang.name} · ${lang.level}", fontSize = 12.sp, color = Color(0xFF444444))
                        }
                    }
                }
            }
        }

        Spacer(Modifier.height(16.dp))

        // Export button
        Button(
            onClick = onExportPdf,
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = 20.dp)
                .height(52.dp),
            colors = ButtonDefaults.buttonColors(containerColor = Gold),
            shape = RoundedCornerShape(12.dp)
        ) {
            Text("Exporter en PDF", color = DeepBlack, fontWeight = FontWeight.Bold, fontSize = 16.sp)
        }

        Spacer(Modifier.height(32.dp))
    }
}

@Composable
fun CvSection(title: String, content: @Composable () -> Unit) {
    Column(modifier = Modifier.fillMaxWidth().padding(bottom = 16.dp)) {
        Text(
            title.uppercase(),
            fontSize = 10.sp,
            fontWeight = FontWeight.Bold,
            color = Color(0xFFC8A96E),
            letterSpacing = 2.sp
        )
        HorizontalDivider(color = Color(0xFFC8A96E), thickness = 1.dp, modifier = Modifier.padding(vertical = 6.dp))
        content()
    }
}

@Composable
fun CvChip(text: String) {
    Text(text, fontSize = 11.sp, color = Color(0xFFCCCCCC))
}

@Composable
fun SkillBar(skill: Skill, modifier: Modifier = Modifier) {
    Column(modifier = modifier) {
        Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
            Text(skill.name, fontSize = 12.sp, color = Color(0xFF333333))
            Text(skill.level.label, fontSize = 10.sp, color = Color(0xFF888888))
        }
        Spacer(Modifier.height(3.dp))
        LinearProgressIndicator(
            progress = { skill.level.value },
            modifier = Modifier.fillMaxWidth().height(3.dp),
            color = Color(0xFFC8A96E),
            trackColor = Color(0xFFDDDDDD)
        )
    }
}

File: flutter_app/lib/screens/steps/preview_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';

import '../../models/cv_data.dart';
import '../../state/cv_model.dart';
import '../../theme/colors.dart';

/// Port de `ui/screens/PreviewScreen.kt` — rendu écran du CV « papier ».
class PreviewScreen extends StatelessWidget {
  const PreviewScreen({super.key, required this.onExportPdf});

  final VoidCallback onExportPdf;

  @override
  Widget build(BuildContext context) {
    final data = context.watch<CvModel>().cvData;
    final info = data.personalInfo;

    return SingleChildScrollView(
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.stretch,
        children: [
          // Carte blanche simulant une A4
          Container(
            margin: const EdgeInsets.all(12),
            padding: const EdgeInsets.all(24),
            decoration: BoxDecoration(
              color: AppColors.cream,
              borderRadius: BorderRadius.circular(8),
            ),
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.stretch,
              children: [
                // Bloc en-tête
                Container(
                  padding: const EdgeInsets.all(20),
                  decoration: BoxDecoration(
                    color: AppColors.deepBlack,
                    borderRadius: BorderRadius.circular(6),
                  ),
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      Text(
                        info.fullName.isEmpty ? 'Votre Nom' : info.fullName,
                        style: const TextStyle(
                          fontSize: 22,
                          fontWeight: FontWeight.bold,
                          color: AppColors.cream,
                        ),
                      ),
                      Text(
                        info.jobTitle.isEmpty
                            ? 'Titre professionnel'
                            : info.jobTitle,
                        style: const TextStyle(
                          fontSize: 14,
                          color: AppColors.gold,
                          fontWeight: FontWeight.w500,
                        ),
                      ),
                      const SizedBox(height: 8),
                      Wrap(
                        spacing: 16,
                        children: [
                          if (info.email.isNotEmpty) _CvChip(info.email),
                          if (info.phone.isNotEmpty) _CvChip(info.phone),
                          if (info.city.isNotEmpty) _CvChip(info.city),
                        ],
                      ),
                    ],
                  ),
                ),

                const SizedBox(height: 16),

                if (info.summary.isNotEmpty)
                  _CvSection(
                    title: 'Profil',
                    child: Text(
                      info.summary,
                      style: const TextStyle(
                        fontSize: 13,
                        color: Color(0xFF333333),
                        height: 1.54,
                      ),
                    ),
                  ),

                if (data.experiences.isNotEmpty)
                  _CvSection(
                    title: 'Expériences',
                    child: Column(
                      crossAxisAlignment: CrossAxisAlignment.stretch,
                      children: [
                        for (final exp in data.experiences)
                          Padding(
                            padding: const EdgeInsets.only(bottom: 12),
                            child: Column(
                              crossAxisAlignment: CrossAxisAlignment.start,
                              children: [
                                Row(
                                  mainAxisAlignment:
                                      MainAxisAlignment.spaceBetween,
                                  children: [
                                    Flexible(
                                      child: Text(
                                        exp.position,
                                        style: const TextStyle(
                                          fontWeight: FontWeight.w600,
                                          fontSize: 13,
                                          color: AppColors.deepBlack,
                                        ),
                                      ),
                                    ),
                                    Text(
                                      exp.period,
                                      style: const TextStyle(
                                        fontSize: 11,
                                        color: Color(0xFF666666),
                                      ),
                                    ),
                                  ],
                                ),
                                Text(
                                  exp.company,
                                  style: const TextStyle(
                                    fontSize: 12,
                                    color: Color(0xFF444444),
                                    fontWeight: FontWeight.w500,
                                  ),
                                ),
                                if (exp.description.isNotEmpty)
                                  Text(
                                    exp.description,
                                    style: const TextStyle(
                                      fontSize: 12,
                                      color: Color(0xFF555555),
                                      height: 1.5,
                                    ),
                                  ),
                              ],
                            ),
                          ),
                      ],
                    ),
                  ),

                if (data.educations.isNotEmpty)
                  _CvSection(
                    title: 'Formation',
                    child: Column(
                      crossAxisAlignment: CrossAxisAlignment.stretch,
                      children: [
                        for (final edu in data.educations)
                          Padding(
                            padding: const EdgeInsets.only(bottom: 10),
                            child: Column(
                              crossAxisAlignment: CrossAxisAlignment.start,
                              children: [
                                Row(
                                  mainAxisAlignment:
                                      MainAxisAlignment.spaceBetween,
                                  children: [
                                    Flexible(
                                      child: Text(
                                        edu.degree,
                                        style: const TextStyle(
                                          fontWeight: FontWeight.w600,
                                          fontSize: 13,
                                          color: AppColors.deepBlack,
                                        ),
                                      ),
                                    ),
                                    Text(
                                      edu.years,
                                      style: const TextStyle(
                                        fontSize: 11,
                                        color: Color(0xFF666666),
                                      ),
                                    ),
                                  ],
                                ),
                                Text(
                                  edu.school,
                                  style: const TextStyle(
                                    fontSize: 12,
                                    color: Color(0xFF444444),
                                  ),
                                ),
                                if (edu.field.isNotEmpty)
                                  Text(
                                    edu.field,
                                    style: const TextStyle(
                                      fontSize: 12,
                                      color: Color(0xFF666666),
                                    ),
                                  ),
                              ],
                            ),
                          ),
                      ],
                    ),
                  ),

                if (data.skills.isNotEmpty)
                  _CvSection(
                    title: 'Compétences',
                    child: Column(
                      children: [
                        for (final row in _chunked(data.skills, 2))
                          Padding(
                            padding: const EdgeInsets.only(bottom: 8),
                            child: Row(
                              children: [
                                for (final skill in row) ...[
                                  Expanded(child: _SkillBar(skill: skill)),
                                  const SizedBox(width: 12),
                                ],
                                if (row.length == 1) const Spacer(),
                              ],
                            ),
                          ),
                      ],
                    ),
                  ),

                if (data.languages.isNotEmpty)
                  _CvSection(
                    title: 'Langues',
                    child: Wrap(
                      spacing: 16,
                      children: [
                        for (final lang in data.languages)
                          Text(
                            '${lang.name} · ${lang.level}',
                            style: const TextStyle(
                              fontSize: 12,
                              color: Color(0xFF444444),
                            ),
                          ),
                      ],
                    ),
                  ),
              ],
            ),
          ),

          const SizedBox(height: 16),

          Padding(
            padding: const EdgeInsets.symmetric(horizontal: 20),
            child: SizedBox(
              height: 52,
              child: FilledButton(
                style: FilledButton.styleFrom(
                  backgroundColor: AppColors.gold,
                  foregroundColor: AppColors.deepBlack,
                  shape: RoundedRectangleBorder(
                    borderRadius: BorderRadius.circular(12),
                  ),
                ),
                onPressed: onExportPdf,
                child: const Text(
                  'Exporter en PDF',
                  style: TextStyle(
                    fontWeight: FontWeight.bold,
                    fontSize: 16,
                  ),
                ),
              ),
            ),
          ),

          const SizedBox(height: 32),
        ],
      ),
    );
  }
}

List<List<T>> _chunked<T>(List<T> list, int size) {
  return [
    for (var i = 0; i < list.length; i += size)
      list.sublist(i, (i + size).clamp(0, list.length)),
  ];
}

class _CvSection extends StatelessWidget {
  const _CvSection({required this.title, required this.child});

  final String title;
  final Widget child;

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.only(bottom: 16),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.stretch,
        children: [
          Text(
            title.toUpperCase(),
            style: const TextStyle(
              fontSize: 10,
              fontWeight: FontWeight.bold,
              color: Color(0xFFC8A96E),
              letterSpacing: 2,
            ),
          ),
          const Padding(
            padding: EdgeInsets.symmetric(vertical: 6),
            child: Divider(height: 1, thickness: 1, color: Color(0xFFC8A96E)),
          ),
          child,
        ],
      ),
    );
  }
}

class _CvChip extends StatelessWidget {
  const _CvChip(this.text);

  final String text;

  @override
  Widget build(BuildContext context) {
    return Text(
      text,
      style: const TextStyle(fontSize: 11, color: Color(0xFFCCCCCC)),
    );
  }
}

class _SkillBar extends StatelessWidget {
  const _SkillBar({required this.skill});

  final Skill skill;

  @override
  Widget build(BuildContext context) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.stretch,
      children: [
        Row(
          mainAxisAlignment: MainAxisAlignment.spaceBetween,
          children: [
            Flexible(
              child: Text(
                skill.name,
                style: const TextStyle(
                  fontSize: 12,
                  color: Color(0xFF333333),
                ),
              ),
            ),
            Text(
              skill.level.label,
              style: const TextStyle(fontSize: 10, color: Color(0xFF888888)),
            ),
          ],
        ),
        const SizedBox(height: 3),
        LinearProgressIndicator(
          value: skill.level.value,
          minHeight: 3,
          color: const Color(0xFFC8A96E),
          backgroundColor: const Color(0xFFDDDDDD),
        ),
      ],
    );
  }
}
